### PR TITLE
orders: Add AdjustBaseAmount 

### PR DESCRIPTION
**Pull request by shazbert**
The `AdjustBaseAmount` method alerts if non-fatal change occurs exchange side
This also adjusts BTCM and adds in price to response. 

Fixes # (issue)

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [ ] go test ./... -race
- [ ] golangci-lint run
- [ ] Test X

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: exchanges/btcmarkets/btcmarkets_wrapper.go [hunk 1]
```diff
@@ -592,7 +592,21 @@ func (b *BTCMarkets) SubmitOrder(ctx context.Context, s *order.Submit) (*order.S
...
+
+	submitResp, err := s.DeriveSubmitResponse(tempResp.OrderID)
+	if err != nil {
+		return nil, err
+	}
+
+	err = submitResp.AdjustBaseAmount(tempResp.Amount)
+	if err != nil {
+		return nil, err
+	}
```
Comment (gloriousCode) [inline exchanges/btcmarkets/btcmarkets_wrapper.go:604]: This is incompatible with targetAmount field:
image

image

I would argue returning an error after an order has successfully submitted is not good. It means the order manager/CMD user won't know about it, even if it was successful

File: exchanges/order/orders.go [hunk 2]
```diff
@@ -475,6 +476,62 @@ func (s *Submit) DeriveSubmitResponse(orderID string) (*SubmitResponse, error) {
...
+		return errAmountIsZero
+	}
+
+	if s.Amount == a {
+		return nil
+	}
+
+	// Warning because amounts should conform to exchange requirements prior to
+	// call but this is not fatal.
+	log.Warnf(log.ExchangeSys, "exchange %v has adjusted requested base amount from %v to %v",
```
Comment (gloriousCode) [inline exchanges/order/orders.go:498]: for orderID %v ?

File: exchanges/order/orders.go [hunk 3]
```diff
@@ -475,6 +476,62 @@ func (s *Submit) DeriveSubmitResponse(orderID string) (*SubmitResponse, error) {
...
+		return errAmountIsZero
+	}
+
+	if s.QuoteAmount == a {
+		return nil
+	}
+
+	// Warning because amounts should conform to exchange requirements prior to
+	// call but this is not fatal.
+	log.Warnf(log.ExchangeSys, "exchange %v has adjusted requested quote amount from %v to %v",
```
Comment (gloriousCode) [inline exchanges/order/orders.go:526]: Please include orderID. Less important would be asset and pair depending on exchange

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Can't believe you're forcing me to buy your LTC bags via this PR. Just two minor things

**Review by gloriousCode — APPROVED** [review]
[In reference to this](https://github.com/thrasher-corp/gocryptotrader/pull/1181#discussion_r1187067522) I will get you to add asset and currency when the usage of this is expanded and different assets have different order endpoints and orderIDs
tACK!

File: exchanges/order/orders.go [hunk 4]
```diff
@@ -475,6 +476,64 @@ func (s *Submit) DeriveSubmitResponse(orderID string) (*SubmitResponse, error) {
...
+		return errAmountIsZero
+	}
+
+	if s.Amount == a {
+		return nil
+	}
+
+	// Warning because amounts should conform to exchange requirements prior to
+	// call but this is not fatal.
+	log.Warnf(log.ExchangeSys, "Exchange %s: has adjusted OrderID: %v requested base amount from %v to %v",
```
Comment (thrasher-) [inline exchanges/order/orders.go:498]: Just need to swap the order of s.OrderID and s.Exchange

File: exchanges/order/orders.go [hunk 5]
```diff
@@ -475,6 +476,64 @@ func (s *Submit) DeriveSubmitResponse(orderID string) (*SubmitResponse, error) {
...
+		return errAmountIsZero
+	}
+
+	if s.QuoteAmount == a {
+		return nil
+	}
+
+	// Warning because amounts should conform to exchange requirements prior to
+	// call but this is not fatal.
+	log.Warnf(log.ExchangeSys, "Exchange %s: has adjusted OrderID: %v requested quote amount from %v to %v",
```
Comment (thrasher-) [inline exchanges/order/orders.go:527]: Same goes for here

File: exchanges/btcmarkets/btcmarkets_wrapper.go [hunk 6]
```diff
@@ -592,7 +592,29 @@ func (b *BTCMarkets) SubmitOrder(ctx context.Context, s *order.Submit) (*order.S
...
+
+	submitResp, err := s.DeriveSubmitResponse(tempResp.OrderID)
+	if err != nil {
+		return nil, err
+	}
+
+	if tempResp.Amount != 0 {
+		err = submitResp.AdjustBaseAmount(tempResp.Amount)
+		if err != nil {
+			log.Errorf(log.ExchangeSys, "Exchange %s: OrderID: %s Base Amount Conversion Error: %s\n", b.Name, submitResp.OrderID, err)
```
Comment (thrasher-) [inline exchanges/btcmarkets/btcmarkets_wrapper.go:604]:
```suggestion
			log.Errorf(log.ExchangeSys, "Exchange %s: OrderID: %s base amount conversion error: %s\n", b.Name, submitResp.OrderID, err)
```

File: exchanges/btcmarkets/btcmarkets_wrapper.go [hunk 7]
```diff
@@ -592,7 +592,29 @@ func (b *BTCMarkets) SubmitOrder(ctx context.Context, s *order.Submit) (*order.S
...
+		}
+	}
+
+	if tempResp.TargetAmount != 0 {
+		err = submitResp.AdjustQuoteAmount(tempResp.TargetAmount)
+		if err != nil {
+			log.Errorf(log.ExchangeSys, "Exchange %s: OrderID: %s Quote Amount Conversion Error: %s\n", b.Name, submitResp.OrderID, err)
```
Comment (thrasher-) [inline exchanges/btcmarkets/btcmarkets_wrapper.go:611]:
```suggestion
			log.Errorf(log.ExchangeSys, "Exchange %s: OrderID: %s quote amount conversion error: %s\n", b.Name, submitResp.OrderID, err)
```

**Review by thrasher- — CHANGES_REQUESTED** [review]
Some minor nits, then good to merge!

**Review by thrasher- — APPROVED** [review]
Thanks for making those changes!

**Review by gloriousCode — APPROVED** [review]

**Pull request merged**